rust/actyx/ax/src/cmd/events/publish.rs

```diff
@@ -24,13 +27,45 @@ use util::{
 pub struct PublishOpts {
     #[structopt(flatten)]
     console_opt: ConsoleOpt,
-    /// event payload (JSON)
-    payload: serde_json::Value,
-    #[structopt(long, short)]
+
+    /// Event payload, needs to be valid JSON.
+    /// You may also pass a file in using the syntax `@file.json` or
+    /// have the command read from standard input using `@-`.
+    payload: String,
+
     /// tag (can be given multiple times)
+    #[structopt(long, short)]
     tag: Option<Vec<Tag>>,
 }
 
+fn load_json(input: String) -> ActyxOSResult<serde_json::Value> {
+    let i = if input == "@-" {
+        let stdin = std::io::stdin();
+        let mut stdin = stdin.lock(); // locking is optional
+
+        let mut line = String::new();
+        while let Ok(n_bytes) = stdin.read_to_string(&mut line) {
+            if n_bytes == 0 {
+                break;
+            }
+        }
+        serde_json::from_str(&line)
+    } else if input.starts_with('@') {
+        let f: &str = input
+            .chars()
+            .next()
+            .map(|c| &input[c.len_utf8()..])
+            .ok_or_else(|| anyhow!("Malformed input"))
+            .ax_invalid_input()?;
```

Comment:
given that the Unicode-encoding of `'@'` is unlikely to change, this can be simplified to `let f = &input[1..]`.